src/tox/session/env_select.py

```diff
@@ -241,13 +241,29 @@ def _finalize_config(self) -> None:
         self._state.conf.core.mark_finalized()
 
     def _build_run_env(self, name: str) -> RunToxEnv | None:
-        if self._provision is not None and self._provision[0] is False and name == self._provision[1]:
+        provision_on = provision_tox_env = provision_loader = None
```

Comment:
I think this file needs to be untouched, only the provision.py should be changed.